virttest/utils_nbd.py

```diff
@@ -0,0 +1,194 @@
+        server_credential_dict = {}
+        server_credential_dict['cakey'] = 'ca-key.pem'
+        server_credential_dict['cacert'] = 'ca-cert.pem'
+        server_credential_dict['serverkey'] = 'server-key.pem'
+        server_credential_dict['servercert'] = 'server-cert.pem'
+        # Build a server key.
+        build_server_key(server_cert_dir, tmp_ca_cert_dir,
+                         cn, certtool="certtool",
+                         credential_dict=server_credential_dict, on_local=True)
+
+        server_key_path = os.path.join(server_cert_dir, server_credential_dict['serverkey'])
+
+        # Create server certs and keys
+        client_cert_dir = '/etc/pki/libvirt-nbd'
+        # Clean up old client certs and keys.
+        if os.path.exists(client_cert_dir):
+            shutil.rmtree(client_cert_dir)
+        if not os.path.exists(client_cert_dir):
+            os.makedirs(client_cert_dir)
+        # Copy ca-key.pem and ca-cert.pem into client_cert_dir
+        shutil.copy2(os.path.join(tmp_ca_cert_dir, ca_credential_dict['cakey']), os.path.join(client_cert_dir, ca_credential_dict['cakey']))
+        shutil.copy2(os.path.join(tmp_ca_cert_dir, ca_credential_dict['cacert']), os.path.join(client_cert_dir, ca_credential_dict['cacert']))
+        client_credential_dict = {}
+        client_credential_dict['cakey'] = 'ca-key.pem'
+        client_credential_dict['cacert'] = 'ca-cert.pem'
+        client_credential_dict['clientkey'] = 'client-key.pem'
+        client_credential_dict['clientcert'] = 'client-cert.pem'
+
+        # build a client key.
+        build_client_key(client_cert_dir,
+                         client_cn=cn, certtool="certtool",
+                         credential_dict=client_credential_dict)
+
+    def start_nbd_server(self):
+        # Clean up pre-occupied port if existed.
+        pre_clean = "kill -9 $(ps aux | grep 'qemu-nbd'|grep %s | awk '{print $2}')" % self.port
+        process.run(pre_clean, ignore_status=True, shell=True, verbose=False)
+        # Create image.
+        self._create_img()
+        # Start qemu nbd server.
+        try:
+            tls_object_str = ' '
+            if self.tls:
+                self.setup_certs()
+                self.update_nbd_tls_config()
+                tls_creds = 'libivrt_%s' % self.port
+                tls_object_str = "--object tls-creds-x509,id=%s,endpoint=server,dir=/etc/pki/qemu  --tls-creds=%s" % (tls_creds, tls_creds)
+
+            qemu_nbd_cmd = 'qemu-nbd -t %s -f %s %s -p %s ' % (tls_object_str, self.image_format, self.image, self.port)
+            if self.export_name:
+                qemu_nbd_cmd += "-x %s " % self.export_name
+            qemu_nbd_cmd += "&"
+            process.run(qemu_nbd_cmd, ignore_status=False, shell=True, verbose=True, ignore_bg_processes=True)
+            logging.info("nbd server start at port: %s", self.port)
+        except Exception as info:
+            logging.info("nbd server fail to start")
```

Comment:
Not use "info" in fact

Reply:
this help debug since raise statement may lose backtrace